src/controller/python/chip/commissioning/commissioning_flow_blocks.py

```diff
@@ -0,0 +1,182 @@
+        ))
+
+        self._logger.info("Adding Operational Certificate")
+        response = await self._devCtrl.SendCommand(device.node_id, commissioning.ROOT_ENDPOINT_ID, Clusters.OperationalCredentials.Commands.AddNOC(
+            NOCValue=commissionee_credentials.noc,
+            ICACValue=commissionee_credentials.icac,
+            IPKValue=commissionee_credentials.ipk,
+            caseAdminSubject=commissionee_credentials.case_admin_node,
+            adminVendorId=commissionee_credentials.admin_vendor_id
+        ))
+        if response.statusCode != 0:
+            raise commissioning.CommissionFailure(repr(response))
+
+        self._logger.info("Setting fabric label")
+        response = await self._devCtrl.SendCommand(device.node_id, commissioning.ROOT_ENDPOINT_ID, Clusters.OperationalCredentials.Commands.UpdateFabricLabel(
+            label=parameter.fabric_label
+        ))
+        if response.statusCode != 0:
+            raise commissioning.CommissionFailure(repr(response))
+
+        self._logger.info(f"Device peer id: {commissionee_credentials.fabric_id:016x}:{commissionee_credentials.node_id:016x}")
+
+        return commissionee_credentials.node_id
+
+    async def network_commissioning_thread(self, parameter: commissioning.Parameters, device: pase.Session):
+        if not parameter.thread_credentials:
+            raise TypeError("The device requires a Thread network dataset")
+
+        self._logger.info("Adding Thread network")
+        response = await self._devCtrl.SendCommand(nodeid=device.node_id, endpoint=commissioning.ROOT_ENDPOINT_ID, payload=Clusters.NetworkCommissioning.Commands.AddOrUpdateThreadNetwork(
+            operationalDataset=parameter.thread_credentials))
+        if response.networkingStatus != Clusters.NetworkCommissioning.Enums.NetworkCommissioningStatus.kSuccess:
+            raise commissioning.CommissionFailure(f"Unexpected result for adding network: {response.networkingStatus}")
+
+        network_list = (await self._devCtrl.ReadAttribute(nodeid=device.node_id, attributes=[(commissioning.ROOT_ENDPOINT_ID, Clusters.NetworkCommissioning.Attributes.Networks)], returnClusterObject=True))[commissioning.ROOT_ENDPOINT_ID][Clusters.NetworkCommissioning].networks
+        network_id = network_list[response.networkIndex].networkID
+
+        self._logger.info("Enabling Thread network")
+        response = await self._devCtrl.SendCommand(nodeid=device.node_id, endpoint=commissioning.ROOT_ENDPOINT_ID, payload=Clusters.NetworkCommissioning.Commands.ConnectNetwork(networkID=network_id), interactionTimeoutMs=self._devCtrl.ComputeRoundTripTimeout(device.node_id, upperLayerProcessingTimeoutMs=30000))
+        if response.networkingStatus != Clusters.NetworkCommissioning.Enums.NetworkCommissioningStatus.kSuccess:
+            raise commissioning.CommissionFailure(f"Unexpected result for enabling network: {response.networkingStatus}")
+
+        self._logger.info(f"Thread network commissioning finished")
```

Comment:
We have to be careful here for how we will support non-concurrent commissioning.